app/backend/src/couchers/servicers/communities.py

```diff
@@ -245,3 +245,83 @@ def LeaveCommunity(self, request, context):
+                .filter(Node.id >= next_node_id)
+                .order_by(Node.id)
+                .limit(page_size + 1)
+                .all()
+            )
+            
+            return communities_pb2.ListUserCommunitiesRes(
+                communities=[community_to_pb(node, user_id) for node in nodes[:page_size]],
+                next_page_token=str(nodes[-1].id) if len(nodes) > page_size else None,
+            )
+    
+    def ListUserGroups(self, request, context):
+        with session_scope() as session:
+            page_size = min(MAX_PAGINATION_LENGTH, request.page_size or MAX_PAGINATION_LENGTH)
+            next_cluster_id = int(request.page_token) if request.page_token else 0
+            user_id = context.user_id if request.user_id == 0 else request.user_id
+            clusters = (
+                session.query(Cluster)
+                .join(ClusterSubscription, ClusterSubscription.cluster_id == Cluster.id)
+                .filter(ClusterSubscription.user_id == user_id)
+                .filter(~Cluster.is_official_cluster)  # not an official group
+                .filter(Cluster.id >= next_cluster_id)
+                .order_by(Cluster.id)
+                .limit(page_size + 1)
+                .all()
+            )
+            return communities_pb2.ListUserGroupsRes(
+                groups=[group_to_pb(cluster, user_id) for cluster in clusters[:page_size]],
+                next_page_token=str(clusters[-1].id) if len(clusters) > page_size else None,
+            )
+    
+    def ListUserPlaces(self, request, context):
+        with session_scope() as session:
+            page_size = min(MAX_PAGINATION_LENGTH, request.page_size or MAX_PAGINATION_LENGTH)
+            next_page_id = int(request.page_token) if request.page_token else 0
+            user_id = context.user_id if request.user_id == 0 else request.user_id
+            places = (
+                session.query(Page)
+                .filter(Page.owner_user_id == user_id)
+                .filter(Page.type == PageType.place)
+                .filter(Page.id >= next_page_id)
+                .order_by(Page.id)
+                .limit(page_size + 1)
+                .all()
+            )
+            return communities_pb2.ListUserPlacesRes(
+                places=[page_to_pb(page, context.user_id) for page in places[:page_size]],
+                next_page_token=str(places[-1].id) if len(places) > page_size else None,
+            )
+
+    def ListUserGuides(self, request, context):
+        with session_scope() as session:
+            page_size = min(MAX_PAGINATION_LENGTH, request.page_size or MAX_PAGINATION_LENGTH)
+            next_page_id = int(request.page_token) if request.page_token else 0
+            user_id = context.user_id if request.user_id == 0 else request.user_id
+            guides = (
+                session.query(Page)
+                .filter(Page.owner_user_id == user_id)
+                .filter(Page.type == PageType.guide)
```

Comment:
Places and Guides untested. Original ListPlaces and ListGuides were untested with the comment:
# TODO: requires transferring of content

Not too sure what that means

Reply:
Places and guides are currently created by users, then transferred to communities/groups. So communities couldn't have any places/groups before transfer was implemented.

Transfer has been implemented now, but tests haven't been written 🤔